File: benchmarks/benchmark.py
# Python Benchmark: recursive fib(28) and loop of 100,000
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

print("--- BENCHMARK PYTHON ---")

# 1. Fibonacci
fib_start = time.perf_counter()
fib_result = fib(28)
fib_duration = (time.perf_counter() - fib_start) * 1000
print("1. Fibonacci(28) recursivo:")
print(f"   Resultado: {fib_result}")
print(f"   Tiempo: {fib_duration:.2f} ms")

# 2. Loop
loop_start = time.perf_counter()
count = 0
for _ in range(100000):
    count += 1
loop_duration = (time.perf_counter() - loop_start) * 1000
print("2. Bucle simple de 100,000 iteraciones:")
print(f"   Resultado: {count}")
print(f"   Tiempo: {loop_duration:.2f} ms")

File: benchmarks/benchmark.cs
// C# Benchmark: recursive fib(28) and loop of 100,000
using System;
using System.Diagnostics;

Console.WriteLine("--- BENCHMARK C# ---");

// 1. Fibonacci
var fibStopwatch = Stopwatch.StartNew();
long fibResult = Fib(28);
fibStopwatch.Stop();
Console.WriteLine("1. Fibonacci(28) recursivo:");
Console.WriteLine($"   Resultado: {fibResult}");
Console.WriteLine($"   Tiempo: {fibStopwatch.Elapsed.TotalMilliseconds:F2} ms");

// 2. Loop
var loopStopwatch = Stopwatch.StartNew();
int count = 0;
for (int i = 0; i < 100000; i++) {
    count += 1;
}
loopStopwatch.Stop();
Console.WriteLine("2. Bucle simple de 100,000 iteraciones:");
Console.WriteLine($"   Resultado: {count}");
Console.WriteLine($"   Tiempo: {loopStopwatch.Elapsed.TotalMilliseconds:F2} ms");

static long Fib(long n) {
    if (n <= 1) return n;
    return Fib(n - 1) + Fib(n - 2);
}
